client/rest-high-level/src/main/java/org/elasticsearch/client/AsyncSearchClient.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.client;
+
+import org.elasticsearch.action.ActionListener;
+import org.elasticsearch.client.asyncsearch.SubmitAsyncSearchRequest;
+import org.elasticsearch.client.asyncsearch.AsyncSearchResponse;
+
+import java.io.IOException;
+
+import static java.util.Collections.emptySet;
+
+public class AsyncSearchClient {
+    private final RestHighLevelClient restHighLevelClient;
+
+    AsyncSearchClient(RestHighLevelClient restHighLevelClient) {
+        this.restHighLevelClient = restHighLevelClient;
+    }
+
+    // TODO add docs url
```

Comment:
docs link should be https://www.elastic.co/guide/en/elasticsearch/reference/current/async-search.html PR is open but not merged yet: #53675

Reply:
done, see ebd1906